### ISSUE TITLE
Docs: Improve search functionality

### ISSUE THREAD

**Issue by mbercx** [aiidateam/aiida-core#6379]
### Describe the current issue

We have quite a bit of documentation, which is great, but often it is difficult for a user to find the documentation they are interested in. For example:

![image](https://github.com/aiidateam/aiida-core/assets/25607960/106806b1-6aba-49c5-9bb3-065a1d7069a2)

The first hit here is an `index.html` file. Often you will also find API references among the top hits.

### Describe the solution you'd like

Look for ways (if any) to improve which content is presented by the search functionality. One suggestion by @giovannipizzi:

https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_search_scorer

**Comment by danielhollas** [comment]
Yes please, what you describe has been my experience as well. The low hanging fruit would be to put API references always at the bottom of search results.

**Comment by killercatempire** [comment]
Would it be possible to use the PageRank algorithm? That is what google uses for their search engine. The amount of documentation is small enough that it would be fairly cheap and quick
